java/arcs/android/devtools/DevToolsService.kt

```diff
@@ -54,4 +69,39 @@ class DevToolsService : Service() {
         devToolsServer.close()
         scope.cancel()
     }
+
+    suspend fun initialize() {
+        check(
+            serviceConnection == null ||
+            storageService == null ||
+            storageService?.asBinder()?.isBinderAlive != true
+        ) {
+            "Connection to StorageService is already alive."
+        }
+        val connectionFactory = DevToolsConnectionFactory(
+            this@DevToolsService,
+            getStorageClass()
+        )
+        this.serviceConnection = connectionFactory()
+        // Need to initiate the connection on the main thread.
+        this.storageService = IDevToolsStorageManager.Stub.asInterface(
+            serviceConnection?.connectAsync()?.await()
+        )
+    }
+
+    private fun getStorageClass(): Class<StorageService> {
+        val storageClass: Class<StorageService> = StorageService::class.java
+        val packageManager: PackageManager = packageManager
+        val installedPackages: List<PackageInfo> = packageManager
+            .getInstalledPackages(PackageManager.GET_SERVICES)
+        for (packageInfo in installedPackages) {
+            val services: Array<ServiceInfo>? = packageInfo.services
+            services?.forEach { service ->
+                if (service.javaClass.superclass == storageClass) {
+                    service.javaClass
+                }
+            }
+        }
+        return storageClass
+    }
```

Comment:
I don't see how this is ever updating `storageClass` (in fact: it's a `val`, so it couldn't be updated anyway)...

Reply:
Oops, I meant for line 101 to be `return service.javaClass`. But when I switched to this, it didn't work as it return the ServiceInfo class. I've changed it now so the developer has to pass the subclass of StorageService via a bundled intent. While this isn't as easy for the developer, it seems like if they are subclassing StorageService they are a "power user" and this isn't a huge burden.